static/src/javascripts/projects/common/modules/onward/popular.js

```diff
@@ -44,7 +46,11 @@ define([
     };
 
     MostPopular.prototype.prerender = function () {
-        if (!(config.page.shouldHideAdverts || this.mobileMaximumSlotsReached())) {
+        if (!(
+            config.page.shouldHideAdverts || /* sensitive pages */
```

Comment:
@Calanthe We don't simply remove the elements; we never create the ad slots in the first place. When DFP queries the document for MPU candidates, they won't get matched. So no network calls should be made for them.

@davidrapson Yeah, it's awkward, but the behaviour of adfree and sensitive pages are not exaclty the same. Some components - such as merchandising and outbrain - are going to remain active, at least in the first iteration. I do agree the logic that defines what ad-free actually _is_ is smeared across several modules, though. I did think about perhaps creating a map of flags that would define which features would and wouldn't be turned on, and would let us define ad-free and sensitive-mode behaviour in fairly self-documenting ways, but wasn't sure it'd be worthwhile when there weren't many permutations of ad behaviours, and sensitive mode touches lots of non-ad things too.